src/dbnode/encoding/iterators.go

```diff
@@ -33,17 +34,18 @@ var (
 	// use max int64 for the seconds component only then integer overflow
 	// will occur when performing comparisons like time.Before() and they
 	// will not work correctly.
-	timeMax = time.Unix(1<<63-62135596801, 999999999)
+	timeMax      = time.Unix(1<<63-62135596801, 999999999)
+	timeMaxNanos = int64(math.MaxInt64)
 )
 
 // iterators is a collection of iterators, and allows for reading in order values
 // from the underlying iterators that are separately in order themselves.
 type iterators struct {
 	values             []Iterator
 	earliest           []Iterator
-	earliestAt         time.Time
-	filterStart        time.Time
-	filterEnd          time.Time
+	earliestAt         int64
+	filterStart        int64
+	filterEnd          int64
```

Comment:
Can you use xtime.UnixNanos here? (the xtime package is at `github.com/m3db/m3/src/x/time`)